### PR TITLE
Add Pencil 3.0.4

### PR DESCRIPTION
**Pull request by silverkorn**
https://pencil.evolus.vn/  
_(https://github.com/evolus/pencil)_

Pencil is built for the purpose of providing a free and open-source GUI prototyping tool that people can easily install and use to create mockups in popular desktop platforms.

Successfully tested locally with both 64-bit and 32-bit.
Might need a quick check with the hashing for the `autoupdate` part. 
Otherwise, the pattern worked properly on a RegExp sandbox <sup>(https://regex101.com/)</sup>.

### CODE REVIEW

**Comment by linsui** [comment]
```
    "checkver": "Pencil-Setup-([\\d.]+)\\.exe",
    "autoupdate": {
        "url": "https://pencil.evolus.vn/dl/V$version/Pencil-Setup-$version.exe#/dl.7z",
        "hash": {
            "url": "$baseurl/CHECKSUM"
        }
    }
```

Please put post_install before checkver.

**Comment by linsui** [comment]
It's dual licensed. I'm not sure what identifier should be used.

```
    "license": {
        "identifier": "GPL-2.0-only|Freeware",
        "url": "https://pencil.evolus.vn/Licensing.html"
    },
```

File: bucket/pencil.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{
+    "version": "3.0.4",
+    "description": "An open-source GUI prototyping tool available for ALL platforms.",
+    "homepage": "https://pencil.evolus.vn/",
+    "license": "GPL-2.0-only",
+    "url": "https://pencil.evolus.vn/dl/V3.0.4/Pencil-Setup-3.0.4.exe#/dl.7z",
+    "hash": "sha1:f7763ceb6a48645c09d868d58016ac113b484241",
+    "post_install": "Remove-Item \"$dir\\`$PLUGINSDIR\" -Recurse",
```
Comment (linsui) [inline bucket/pencil.json]:
```suggestion
    "post_install": "Remove-Item \"$dir\\`$PLUGINSDIR\", \"$dir\\resources\", \"$dir\\Uninstall Pencil.exe\" -Recurse",
```

**Comment by silverkorn** [comment]
Updates with the fixed manifest.
Please note that it's an Electron application and it needs the content of of the `resources` directory, specially the `.asar` files.

**Pull request merged**